### DISCUSSION TITLE
Revisions in patch

### DISCUSSION THREAD

**Discussion by djordjedjukic** [ravendb/ravendb#13006] [Q&A]
Is it possible to use revisions in the **patch** command?

**Comment by ayende** [comment]
I'm not sure that I'm following. What is it that you are trying to do?
A `put()` in patch will generate a revision, and will modifying the document directly.

If you want to *read* a revision, that is not exposed in the JS API

**Reply by djordjedjukic** [reply]
Yes, I wanted to read a revision.

I need to update all records from the collection (around 50k).
I need to take the first and last revision for every record and to take **@last-modified** field and store it in the current document.

I am trying to find the most efficient way to do that because the limitation is 30 requests per session.

**Reply by ayende** [reply]
You can do an `include` on the revision, and if really needed, you can increase the session limit.

**Reply by djordjedjukic** [reply]
I didn't know I can include revisions, I will check that. 
Thank you.

**Reply by djordjedjukic** [reply]
I tried to include revisions, but I can not figure out how to use that.
Do you have some examples to provide?

I saw some in ravendb tests, but I didn't manage to make them work.

**Reply by ayende** [reply]
@omerlewitz Can you take a look?

**Reply by omerlewitz** [reply]
I attached a sample test that demonstrates how to perform the desired operation.
```
        public void Test()
        {
            using (var store = GetDocumentStore())
            {
                List<string> ids;
                using (var session = store.OpenSession())
                {
                    ids = session.Query<Order>().Select(x => x.Id).ToList();
                }

                foreach (var id in ids)
                {
                    using (var session = store.OpenSession())
                    {
                        var metadatas = session.Advanced.Revisions.GetMetadataFor(id, 0, int.MaxValue); // will bring all the revisions metadata
                        var firstLastModified = metadatas.First().GetString(Constants.Documents.Metadata.LastModified);
                        var lastModified = metadatas.Last().GetString(Constants.Documents.Metadata.LastModified);

                        firstLastModified = Convert.ToDateTime(firstLastModified).EnsureUtc().GetDefaultRavenFormat();
                        lastModified = Convert.ToDateTime(firstLastModified).EnsureUtc().GetDefaultRavenFormat();

                        session.Advanced.Patch<Order, string>(id, x => x.FirstLastModified, firstLastModified);
                        session.Advanced.Patch<Order, string>(id, x => x.LastLastModified, lastModified);
                        
                        session.SaveChanges();
                    }
                }
            }
        }
```
